lalrpop-test/src/util/tok.rs

```diff
@@ -13,10 +13,21 @@ pub enum Tok {
     Times,
     Div,
     Comma,
+    Open(Delim),
+    Close(Delim),
+    String(&'static str),
```

Comment:
This could just be a `String`, no? It may just be for testing but I'd still prefer not to leak the `String`.

Reply:
You can't pattern match on `String`, can you? It needs to be `&str`. For example:

```rust
enum Tok {
    String(String),
    Str(&'static str),
}

fn main() {
    let t1 = Tok::String("x".to_owned());
    let t2 = Tok::Str("x");
    
    // if let Tok::String("x") = t1 {}
    if let Tok::Str("x") = t2 {}
}
```

If you enable the commented-out code this doesn't compile.

I still tried to make this a `String` and it indeed failed with errors like

```
error[E0308]: mismatched types
    --> /home/omer/rust/lalrpop/target/debug/build/lalrpop-test-f7207b3c7f924ac7/out/nested.rs:1464:33
     |
1461 |                 match *__token {
     |                       -------- this expression has type `util::tok::Tok`
...
1464 |                     Tok::String("hello") if true => Some(2),
     |                                 ^^^^^^^ expected struct `std::string::String`, found `&str`
```

Reply:
We could also add a `'input` lifetime to the `Tok` type, but I thought that would be unnecessarily complex and would require refactoring a lot of existing tests.

Reply:
Right, I guess it should have the `'input` lifetime actually. May cause some compilation errors in other places using `Tok` though but should be fixable.